apis/v1beta1/types.go

```diff
@@ -0,0 +1,120 @@
+/*
+Copyright 2024 The Crossplane Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package v1beta1
```

Comment:
Adding a v1beta1 for ProviderConfig lets us remove the BasePackage overrides.
https://github.com/crossplane-contrib/provider-jet-equinix/pull/59/files#diff-483ebdb1139323e5256859b870054a99a208f4a1d60a0f5e51583c22d8600240L95-L104

I don't know of any other benefits to introducing this.